acceptance/tests/vault/helpers.go

```diff
@@ -0,0 +1,180 @@
+	// then enables the Vault agent init container to call 'vault login'
+	// with the Kubernetes auth method to obtain a Vault token.
+	// Please see https://www.vaultproject.io/docs/auth/kubernetes#configuration
+	// for more details.
+	logger.Log(t, "Creating the consul-server and consul-client roles")
+	params := map[string]interface{}{
+		"bound_service_account_names":      consulClientServiceAccountName,
+		"bound_service_account_namespaces": ns,
+		"policies":                         "consul-gossip",
+		"ttl":                              "24h",
+	}
+	_, err := vaultClient.Logical().Write(fmt.Sprintf("auth/%s/role/consul-client", authPath), params)
+	require.NoError(t, err)
+
+	params = map[string]interface{}{
+		"bound_service_account_names":      consulServerServiceAccountName,
+		"bound_service_account_namespaces": ns,
+		"policies":                         fmt.Sprintf("consul-gossip,connect-ca,consul-server-%s", datacenter),
+		"ttl":                              "24h",
+	}
+	_, err = vaultClient.Logical().Write(fmt.Sprintf("auth/%s/role/consul-server", authPath), params)
+	require.NoError(t, err)
+
+	// Create the CA role that all components will use to fetch the Server CA certs.
+	params = map[string]interface{}{
+		"bound_service_account_names":      "*",
+		"bound_service_account_namespaces": ns,
+		"policies":                         "consul-ca",
+		"ttl":                              "24h",
+	}
+	_, err = vaultClient.Logical().Write(fmt.Sprintf("auth/%s/role/consul-ca", authPath), params)
+	require.NoError(t, err)
+}
+
+// configurePKICA generates a CA in Vault.
+func configurePKICA(t *testing.T, vaultClient *vapi.Client) {
+	// Create root CA to issue Consul server certificates and the `consul-server` PKI role.
+	// See https://learn.hashicorp.com/tutorials/consul/vault-pki-consul-secure-tls.
+	// Generate the root CA.
+	params := map[string]interface{}{
+		"common_name": "Consul CA",
+		"ttl":         "24h",
+	}
+	_, err := vaultClient.Logical().Write("pki/root/generate/internal", params)
+	require.NoError(t, err)
+
+	err = vaultClient.Sys().PutPolicy("consul-ca", caPolicy)
+	require.NoError(t, err)
+}
+
+// configurePKICertificates configures roles so that Consul server TLS certificates
+// can be issued by Vault.
+func configurePKICertificates(t *testing.T, vaultClient *vapi.Client, consulReleaseName, ns, datacenter string) string {
+	// Create the Vault PKI Role.
+	consulServerDNSName := consulReleaseName + "-consul-server"
+	allowedDomains := fmt.Sprintf("%s.consul,%s,%s.%s,%s.%s.svc", datacenter, consulServerDNSName, consulServerDNSName, ns, consulServerDNSName, ns)
+	params := map[string]interface{}{
+		"allowed_domains":    allowedDomains,
+		"allow_bare_domains": "true",
+		"allow_localhost":    "true",
+		"allow_subdomains":   "true",
+		"generate_lease":     "true",
+		"max_ttl":            "1h",
+	}
+
+	pkiName := fmt.Sprintf("consul-server-%s", datacenter)
```

Comment:
nit:  does it make sense to call this pkiRoleName to be a bit more descriptive?

Reply:
yeah, that would make more sense!